src/cat.js

```diff
@@ -37,6 +43,34 @@ function _cat(options, files) {
     cat += fs.readFileSync(file, 'utf8');
   });
 
+  if (options.number) {
+    cat = addNumbers(cat);
+  }
+
   return cat;
 }
 module.exports = _cat;
+
+function addNumbers(cat) {
+  var addEOF = false;
+  var lines = cat.split('\n');
+
+  if (lines[lines.length - 1] === '') {
+    addEOF = true;
```

Comment:
Hmmm can we do something better here? It's kind of silly to strip the last array element simply to add it back at the end.

I'm not certain what the best approach is here.

Reply:
I think we can read by chunks from file stream. But even 'readline' package `split` the buffer in lines.
https://github.com/nodejs/node/blob/master/lib/readline.js#L980